### ISSUE TITLE
Base64.strict_decode64 isn't strict

### ISSUE THREAD

**Issue by blinsay** [jruby/jruby#2551]
JRuby's `Base64.strict_decode64` doesn't seem to actually be strict. It's easy to reproduce in a REPL.

```
[5] pry(main)> d = Base64.strict_encode64("hi there")
=> "aGkgdGhlcmU="
[6] pry(main)> Base64.strict_decode64("aGkgdGhlcmU")
=> "hi the"
```

Using IRB in Ruby 2.0p481, the method actually throws an `invalid base64` error.

```
irb(main):007:0> Base64.strict_encode64("hi there")
=> "aGkgdGhlcmU="
irb(main):008:0> Base64.strict_decode64("aGkgdGhlcmU")
ArgumentError: invalid base64
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/base64.rb:73:in `unpack'
	from /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/base64.rb:73:in `strict_decode64'
	from (irb):8
	from /usr/bin/irb:12:in `<main>'
```

I took a quick peek, and the [Ruby source](https://github.com/ruby/ruby/blob/v2_0_0_481/lib/base64.rb#L72-L74) in the `Base64` modules of both versions looks identical. Didn't have time to dive into `pack` to see what was going on.

**Comment by rtyler** [comment]
Which version of JRuby are you working with in the first version?

**Comment by blinsay** [comment]
It's happening on 1.7.9, 1.7.16, and 9000-pre.

```
[12:44:00] benl:jruby-testing $ rbenv local jruby-1.7.9
[12:44:09] benl:jruby-testing $ ruby base64_strict_test.rb
test string = 'hi there'
encoded = aGkgdGhlcmU=
Base64.strict_decode64(aGkgdGhlcmU)
decoded = 'hi the'
[12:44:12] benl:jruby-testing $ rbenv local jruby-1.7.16
[12:44:15] benl:jruby-testing $ ruby base64_strict_test.rb
test string = 'hi there'
encoded = aGkgdGhlcmU=
Base64.strict_decode64(aGkgdGhlcmU)
decoded = 'hi the'
[12:44:18] benl:jruby-testing $ rbenv local jruby-9.0.0.0.pre1
[12:44:27] benl:jruby-testing $ ruby base64_strict_test.rb
test string = 'hi there'
encoded = aGkgdGhlcmU=
Base64.strict_decode64(aGkgdGhlcmU)
decoded = 'hi the'
```

**Comment by mkristian** [comment]
fixed with 13ff7bf0774b4a6a9af1a5866449667d1cc00cbd

**Comment by blinsay** [comment]
awesome! thanks!
